dig_test.go

```diff
@@ -1403,4 +1403,60 @@ func TestInvokeFailures(t *testing.T) {
 		assert.Contains(t, err.Error(), "dig.in embeds *dig.In")
 		assert.Contains(t, err.Error(), "embed dig.In value instead")
 	})
+
+	t.Run("requesting a value or pointer when other is present", func(t *testing.T) {
+		type A struct{}
+		type outA struct {
+			Out
+
+			A `name:"hello"`
+		}
+		type B struct{}
+
+		cases := []struct {
+			name        string
+			provide     interface{}
+			invoke      interface{}
+			errContains []string
+		}{
+			{
+				name:        "value missing, pointer present",
+				provide:     func() *A { return &A{} },
+				invoke:      func(A) {},
+				errContains: []string{"dig.A is not in the container, did you mean to use *dig.A"},
```

Comment:
This is phenomenal.

Reply:
Might be worth adding a `?` to the end of the err.

Reply:
Totally agree, in fact I included it :) The question mark is a few words later:

`type *dig.A is not in the container, did you mean to use dig.A that is present?`

I wanted to signal that the second type is present in the container. I could use some help with this copy if you guys have better ideas